### DISCUSSION TITLE
General Questions

### DISCUSSION THREAD

**Discussion by jt4000** [elsa-workflows/elsa-core#880] [Q&A]
Hello,

I still don't understand about Elsa.

1. What is the purpose to use Elsa? Any conrete business example?

2. What is the benefit to use Elsa?

3. When is it relevant ot use Elsa?

4. Why you should use Elsa?

5. What context is relevant to use Elsa?

Thank you!

**Comment by sfmskywalker** [comment]
Thanks for the great questions!

1. The purpose of Elsa is to enable .NET developers to add workflow capabilities to their own applications easily. A good example is Orchard Core CMS, which is an open source, multi-tenant, modular content management system that also hosts a workflows module. This module enables users to implement logic that responds to certain events. A classical example would be the "content created" event, in response to which the user wants to request approval by some site moderator to review & approve (or request) changes. The user can implement this sort of thing visually using a workflow. Although Orchard Core doesn't use Elsa, it could have. Similarly, your own application could allow your users to implement logic that responds to a variety of events happening in your system, without requiring them to program in C#.
2. Elsa's most important differentiators compared to other workflow engine frameworks are:

   - Free & open source (BSD-3)
   - Provides a reusable web-based designer that can be integrated into your own web application.
   - Horizontally scalable across multiple nodes within a cluster.
   - Built for .NET Core

More specifically, it's the combination of the above that sets Elsa apart from the rest.
3. Elsa is relevant for a variety of use cases, ranging from the need to [build a simple background task](https://dev.to/sfmskywalker/scheduled-background-tasks-made-easy-with-elsa-workflows-3o6k) to providing workflow capabilities to your application.
4. You might consider using Elsa when:

   - You need a free & open source workflow engine that supports long-running processes, and/or:
   - When you need a workflow designer to create worklows, and/or:
   - When you need a workflow engine that can scale horizontally, and/or:
   - When you are building a .NET application that contains long-running processes.

5. Elsa can be used in roughly the following contexts:
   - Within your own application app domain (in-process)
   - As a service (out-of-process)

I hope this clarifies things a little bit, but feel free to follow up with more questions.
